ReactAndroid/src/main/java/com/facebook/react/animated/FrameBasedAnimationDriver.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.react.animated;
+
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+
+/**
+ * Implementation of {@link AnimationDriver} which provides a support for simple time-based
+ * animations that are pre-calculate on the JS side. For each animation frame JS provides a value
+ * from 0 to 1 that indicates a progress of the animation at that frame.
+ */
+class FrameBasedAnimationDriver extends AnimationDriver {
+
+  private long mStartFrameTimeNanos = -1;
+  private final double[] mFrames;
+  private final double mToValue;
+  private double mFromValue;
+
+  FrameBasedAnimationDriver(ReadableMap config) {
+    ReadableArray frames = config.getArray("frames");
+    int numberOfFrames = frames.size();
+    mFrames = new double[numberOfFrames];
+    for (int i = 0; i < numberOfFrames; i++) {
+      mFrames[i] = frames.getDouble(i);
+    }
+    mToValue = config.getDouble("toValue");
+  }
+
+  @Override
+  public void runAnimationStep(long frameTimeNanos) {
+    if (mStartFrameTimeNanos < 0) {
+      mStartFrameTimeNanos = frameTimeNanos;
+      mFromValue = mAnimatedValue.mValue;
+    }
+    long timeFromStartNanos = (frameTimeNanos - mStartFrameTimeNanos);
+    // frames are calculated at 60FPS, to get index by a given time offset from the start of the
+    // animation, we take the time diff in millisecond and divide it by 60 frames per 1000ms.
+    int frameIndex = (int) (timeFromStartNanos / 1000000L * 60L / 1000L);
```

Comment:
You could also treat the frames as 'the value every 16.6 ms' and then interpolate the values between frames based on the exact time (instead of snapping to an exact 60fps frame value)